runtime/src/main/java/org/corfudb/runtime/collections/StreamingMapDecorator.java

```diff
@@ -131,4 +135,25 @@ public void clear() {
     public Set<Entry<K, V>> entrySet() {
         return mapImpl.entrySet();
     }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public List<V> scanAndFilter(Predicate<? super V> valuePredicate) {
+        return mapImpl.values().parallelStream()
+                .filter(valuePredicate)
+                .collect(Collectors.toCollection(ArrayList::new));
+    }
```

Comment:
This is using Java's common pool. I believe this will introduce a regression -- this is why a separate thread pool was introduced.

Reply:
Done.